onnxruntime/core/providers/cuda/cu_inc/common.cuh

```diff
@@ -181,8 +181,11 @@ __device__ __inline__ half2 _Tanh(half2 a) {
   return __float22half2_rn(tmp);
 }
 
-template <typename T>
-__device__ __inline__ T _Pow(T a, T b);
+// Capture permutations of int32/64/float/double
+template <typename T, typename T1>
+__device__ __inline__ T _Pow(T a, T1 b) {
+  return static_cast<T>(pow(static_cast<double>(a), static_cast<double>(b)));
+}
```

Comment:
Any potential accuracy issues with this approach?

Reply:
I think int64 may potentially loose precision. However, its support is needed and CUDA only supports pow(double, double) and powf(float, float). No  mixed types which Pow(12) requires.

---
In reply to: [417016905](https://github.com/microsoft/onnxruntime/pull/3727#discussion_r417016905) [](ancestors = 417016905)

Reply:
I just confirmed with Pytorch code they do the same thing.

---
In reply to: [417529223](https://github.com/microsoft/onnxruntime/pull/3727#discussion_r417529223) [](ancestors = 417529223,417016905)